components/FlexibleHeader/src/private/MDCStatusBarShifter.m

```diff
@@ -56,21 +55,38 @@ @implementation MDCStatusBarShifter {
 
   BOOL _prefersStatusBarHidden;
   MDCStatusBarShifterState _snapshotState;
+
+  CGFloat _originalStatusBarHeight;
 }
 
 - (void)dealloc {
   [_replicaInvalidatorTimer invalidate];
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
 - (instancetype)init {
   self = [super init];
   if (self) {
     _enabled = YES;
     _snapshottingEnabled = YES;
+
+    _originalStatusBarHeight = [UIApplication mdc_safeSharedApplication].statusBarFrame.size.height;
+    [[NSNotificationCenter defaultCenter]
+        addObserver:self
+           selector:@selector(statusBarDidChangeFrame)
+               name:UIApplicationDidChangeStatusBarFrameNotification
+             object:nil];
   }
   return self;
 }
 
+#pragma mark - Notification
+
+- (void)statusBarDidChangeFrame {
+  CGFloat statusBarHeight = [UIApplication mdc_safeSharedApplication].statusBarFrame.size.height;
```

Comment:
Nit: Consider using [`UIApplicationStatusBarFrameUserInfoKey`](https://developer.apple.com/documentation/uikit/uiapplicationdidchangestatusbarframenotification?language=objc) from the notification.